### PR TITLE
EZP-27916: Improved exception message on missing SetupFactory

### PR DESCRIPTION
**Pull request by alongosz**
# Improves [EZP-27916](https://jira.ez.no/browse/EZP-27916)

|   |  |
| ------------- | ------------- |
| Target branch  | **master**  |
| Related PR  | ezsystems/ezplatform-solr-search-engine#103  |

## Description

As the result of a discussion in the ezsystems/ezplatform-solr-search-engine#103 PR, there's no need to refactor anything in this package.

This PR includes minor exception message improvement when `SetupFactory` is missing.

**TODO**:
- [x] Improve missing `SetupFactory` exception message.
- [x] **Remove TMP commit**
- ~Align SolrBundle SetupFactory FQCN references with changes from ezsystems/ezplatform-solr-search-engine#103~
- ~Discuss backporting this to `6.7`~

### CODE REVIEW

**Comment by alongosz** [comment]
To sum up: how about some more explicit error message as a part of this improvement issue?
I remember when I first tried to run integration test I didn't know what was going on - so this should help new developers.

File: eZ/Publish/API/Repository/Tests/BaseTest.php [hunk 1]
```diff
@@ -158,12 +158,19 @@ protected function getSetupFactory()
...
+                    'Missing mandatory setting $_ENV["setupFactory"]. You are trying to run integration tests. Use one of the available phpunit-integration-*.xml configurations'
+                );
             }
 
             $setupClass = $_ENV['setupFactory'];
             if (false === class_exists($setupClass)) {
-                throw new \ErrorException('$_ENV["setupFactory"] does not reference an existing class.');
+                throw new \ErrorException(
+                    sprintf(
+                        '$_ENV["setupFactory"] does not reference an existing class. Class %s does not exist. Did you forget to install a dependency?',
```
Comment (adamwojs) [inline eZ/Publish/API/Repository/Tests/BaseTest.php]: For me `$_ENV["setupFactory"] does not reference an existing class: %s. Did you forget to install a dependency?` is enough to understand the problem, but I'm not the best choice to ask about wording 😉

File: eZ/Publish/API/Repository/Tests/BaseTest.php [hunk 2]
```diff
@@ -158,12 +158,19 @@ protected function getSetupFactory()
     {
         if (null === $this->setupFactory) {
             if (false === isset($_ENV['setupFactory'])) {
-                throw new \ErrorException('Missing mandatory setting $_ENV["setupFactory"].');
+                throw new \ErrorException(
+                    'Missing mandatory setting $_ENV["setupFactory"]. You are trying to run integration tests. Use one of the available phpunit-integration-*.xml configurations'
```
Comment (andrerom) [inline eZ/Publish/API/Repository/Tests/BaseTest.php]: As user must have specified test suite I'm usnure if the last part is needed.

Maybe what you are trying to say is:
`Missing mandatory setting $_ENV["setupFactory"], this should normally be set in the relevant phpunit-integration-*.xml file and refer to a setupFactory for the given StorageEngine/SearchEngine in use`

File: eZ/Publish/API/Repository/Tests/BaseTest.php (hunk 1 above)
Comment (andrerom) [reply eZ/Publish/API/Repository/Tests/BaseTest.php]: `$_ENV["setupFactory"] does not reference an existing class: %s. Did you forget to install an package dependency?`

Comment (alongosz) [reply eZ/Publish/API/Repository/Tests/BaseTest.php]: Fixed in 41e532f

File: eZ/Publish/API/Repository/Tests/BaseTest.php (hunk 2 above)
Comment (alongosz) [reply eZ/Publish/API/Repository/Tests/BaseTest.php]: > As user must have specified test suite I'm usnure if the last part is needed.

Not really :)
First time I tried to run tests I executed
```
./vendor/bin/phpunit eZ/Publish/API/Repository/Tests/
```
by default PHPUnit takes `phpunit.xml.dist` or `phpunit.xml`

However the message you've provided works for me, changed in the fixup 41e532f

**Comment by andrerom** [comment]
Thanks @alongosz :)

**Pull request merged**